atomic_reactor/plugins/pre_download_remote_source.py

```diff
@@ -0,0 +1,54 @@
+"""
+Copyright (c) 2019 Red Hat, Inc
+All rights reserved.
+
+This software may be modified and distributed under the terms
+of the BSD license. See the LICENSE file for details.
+
+Downloads and unpacks the source code archive from Cachito and sets appropriate build args.
+"""
+
+from __future__ import absolute_import
+
+import tarfile
+import tempfile
+
+from atomic_reactor.plugin import PreBuildPlugin
+from atomic_reactor.download import download_url
+
+
+class DownloadRemoteSourcePlugin(PreBuildPlugin):
+    key = 'download_remote_source'
+    is_allowed_to_fail = False
+
+    def __init__(self, tasker, workflow, remote_source_url,
+                 remote_source_build_args=None):
+        """
+        :param tasker: ContainerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param remote_source_url: URL to download source archive from
+        :param remote_source_build_args: dict of container build args
+                                         to be used when building the image
+        """
+        super(DownloadRemoteSourcePlugin, self).__init__(tasker, workflow)
+        self.url = remote_source_url
+        self.buildargs = remote_source_build_args or {}
+
+    def run(self):
+        """
+        Run the plugin.
+        """
+        # Download and unpack the source code
+        dest_dir = tempfile.mkdtemp(prefix='source-')
```

Comment:
Do we need to clean this up at some point later?

Reply:
I could download it to the workdir, maybe that's better.